src/messaging/ExchangeContext.h

```diff
@@ -0,0 +1,318 @@
+     *  @retval  #CHIP_ERROR_INCORRECT_STATE                if the state of the exchange context is incorrect.
+     *  @retval  #CHIP_NO_ERROR                             if the CHIP layer successfully delivered the message up to the
+     *                                                       protocol layer.
+     */
+    CHIP_ERROR HandleMessage(const PacketHeader & packetHeader, const PayloadHeader & payloadHeader, System::PacketBuffer * msgBuf);
+
+    /**
+     *  Handle a received CHIP message on this exchange.
+     *
+     *  @param[in]    packetHeader  A reference to the PacketHeader object.
+     *
+     *  @param[in]    payloadHeader A reference to the PayloadHeader object.
+     *
+     *  @param[in]    msgBuf        A pointer to the PacketBuffer object holding the CHIP message.
+     *
+     *  @param[in]    umhandler     A unsolicited message callback handler.
+     *
+     *  @retval  #CHIP_ERROR_INVALID_ARGUMENT               if an invalid argument was passed to this HandleMessage API.
+     *  @retval  #CHIP_ERROR_INCORRECT_STATE                if the state of the exchange context is incorrect.
+     *  @retval  #CHIP_NO_ERROR                             if the CHIP layer successfully delivered the message up to the
+     *                                                       protocol layer.
+     */
+    CHIP_ERROR HandleMessage(const PacketHeader & packetHeader, const PayloadHeader & payloadHeader, System::PacketBuffer * msgBuf,
+                             ExchangeContext::MessageReceiveFunct umhandler);
+
+    /**
+     *  Search for an existing exchange that the message applies to.
+     *
+     *  @param[in]    packetHeader  A reference to the PacketHeader object.
+     *
+     *  @param[in]    payloadHeader A reference to the PayloadHeader object.
+     *
+     *  @retval  true                                       If a match is found.
+     *  @retval  false                                      If a match is not found.
+     */
+    bool MatchExchange(const PacketHeader & packetHeader, const PayloadHeader & payloadHeader);
+
+    /**
+     *  Set delegate of the current ExchangeContext instance.
+     */
+    void SetDelegate(ExchangeContextDelegate * delegate) { mDelegate = delegate; }
+
+    /**
+     *  Return the delegate pointer of the current ExchangeContext instance.
+     */
+    ExchangeContextDelegate * GetDelegate() { return mDelegate; }
+
+    /**
+     *  Set the current ExchangeManager instance.
+     */
+    void SetExchangeMgr(ExchangeManager * exMgr) { mExchangeMgr = exMgr; }
+
+    /**
+     *  Return the pointer of the current associated ExchangeManager instance.
+     */
+    ExchangeManager * GetExchangeMgr() { return mExchangeMgr; }
+
+    /**
+     *  Set the Node ID of peer node.
+     */
+    void SetPeerNodeId(uint64_t nodeId) { mPeerNodeId = nodeId; }
+
+    /**
+     *  Return the Node ID of peer node.
+     */
+    uint64_t GetPeerNodeId() { return mPeerNodeId; }
```

Comment:
Getters for numbers could be marked const. 

DocComment not needed. GetPeerNodeId is obvious 'returns the peer node id' and 'setExchangeId' obviously 'sets the exchange id'. I complain about these doc comments because I have limited vertical space on my screen: using 3 lines of comments to tell me 'SetExchangeId' does a 'Set exchange ID' is wasteful.

Reply:
Acked